frontends/ast/genrtlil.cc

```diff
@@ -2124,11 +2124,9 @@ RTLIL::SigSpec AstNode::genRTLIL(int width_hint, bool sign_hint)
 						if (sig.is_wire()) {
 							// if the resulting SigSpec is a wire, its
 							// signedness should match that of the AstNode
-							if (arg->type == AST_IDENTIFIER && arg->id2ast && arg->id2ast->is_signed && !arg->is_signed)
-								// fully-sliced signed wire will be resolved
-								// once the module becomes available
-								log_assert(attributes.count(ID::reprocess_after));
-							else
+							// unless this instantiation depends on module
+							// information that isn't available yet
+							if (!attributes.count(ID::reprocess_after))
 								log_assert(arg->is_signed == sig.as_wire()->is_signed);
```

Comment:
So what we are checking here is: does the wire we are connecting have an `is_signed` flag matching the signedness of the expression in the connection?

Reply:
That's correct. Should I clarify this comment?

Reply:
No need I think, I am primarily checking my understanding of what happens here, and maybe leaving some notes for anyone else who would look at the PR.